packages/data/src/types.ts

```diff
@@ -1,11 +1,26 @@
-type MapOf< T > = { [ name: string ]: T };
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { MutableRefObject } from 'react';
+
+export type MapOf< T > = { [ name: string ]: T };
 
 export type ActionCreator = Function | Generator;
 export type Resolver = Function | Generator;
 export type Selector = Function;
 
 export type AnyConfig = ReduxStoreConfig< any, any, any >;
 
+export interface SelectMapper {
+	(
+		select: ( reference: StoreReference ) => MapOf< Function >,
+		registry: DataRegistry
+	): unknown;
+}
+export type SelectChooser = SelectMapper | StoreReference;
```

Comment:
This looks like something very specific to the useSelect hook and not a "meaningful" type. It's just a nitpick and I don't really have experience with typescript much but I'd have avoided a named typed for this one and just inlined it personally.

Reply:
this appears again in the type for `DataRegistry` for the `select` function. I was going to move it but forgot about that one. in time I'd like to come up with some better descriptions and names for these but I think for now at least we need to keep it here or duplicate the definition (which I'd rather avoid) so we don't cause false negatives on registry objects.